libchisel/src/remapstart.rs

```diff
@@ -0,0 +1,269 @@
+use parity_wasm::elements::*;
+use std::collections::HashMap;
+
+use super::{ModuleError, ModulePreset, ModuleTranslator};
+
+pub struct RemapStart;
+
+impl ModulePreset for RemapStart {
+    fn with_preset(preset: &str) -> Result<Self, ()> {
+        match preset {
+            // TODO refactor this later
+            "ewasm" => Ok(RemapStart {}),
+            _ => Err(()),
+        }
+    }
+}
+
+impl ModuleTranslator for RemapStart {
+    fn translate_inplace(&self, module: &mut Module) -> Result<bool, ModuleError> {
+        Ok(remap_start(module))
+    }
+
+    fn translate(&self, module: &Module) -> Result<Option<Module>, ModuleError> {
+        let mut ret = module.clone();
+        if remap_start(&mut ret) {
+            Ok(Some(ret))
+        } else {
+            Ok(None)
+        }
+    }
+}
+
+fn clear_names_section(module: &mut Module) {
+    let sections = module.sections_mut();
+    let mut rmidx = sections.len();
+    for (index, section) in sections.iter_mut().enumerate() {
+        if let Section::Name(_) = section {
+            rmidx = index;
+            break;
+        }
+    }
+    if rmidx < sections.len() {
+        sections.remove(rmidx);
+    }
+}
+
+// replace an exported function with the function pointed to by the start section
+fn replace_export_with_start(module: &mut Module, export_name: &str) -> bool {
```

Comment:
If this is a generic helper I'd rename it to `ensure_export_match_function_index(module: &mut Module, export_name: &str, function_idx: usize)` (usize or whatever the func index's type is)

Reply:
does `replace_export` work?